driver-core/src/main/com/mongodb/internal/capi/MongoCryptHelper.java

```diff
@@ -45,28 +42,9 @@ public static MongoCryptOptions createMongoCryptOptions(final Map<String, Map<St
 
         BsonDocument bsonKmsProviders = new BsonDocument();
         for (Map.Entry<String, Map<String, Object>> entry : kmsProviders.entrySet()) {
-            if (entry.getKey().equals("aws")) {
-                mongoCryptOptionsBuilder.awsKmsProviderOptions(
-                        MongoAwsKmsProviderOptions.builder()
-                                .accessKeyId((String) entry.getValue().get("accessKeyId"))
-                                .secretAccessKey((String) entry.getValue().get("secretAccessKey"))
-                                .build()
-                );
-            } else if (entry.getKey().equals("local")) {
-                mongoCryptOptionsBuilder.localKmsProviderOptions(
-                        MongoLocalKmsProviderOptions.builder()
-                                .localMasterKey(ByteBuffer.wrap((byte[]) entry.getValue().get("key")))
-                                .build()
-                );
-            } else {
-                bsonKmsProviders.put(entry.getKey(), new BsonDocumentWrapper<>(new Document(entry.getValue()), new DocumentCodec()));
-            }
+            bsonKmsProviders.put(entry.getKey(), new BsonDocumentWrapper<>(new Document(entry.getValue()), new DocumentCodec()));
```

Comment:
Much simpler now - the new options will now just pass through to the MongoCrypt library.

Reply:
Yep, very clean now